pkg/images/enricher/enricher_impl.go

```diff
@@ -605,27 +605,30 @@ func (e *enricherImpl) enrichWithSignature(ctx context.Context, enrichmentContex
 	}
 
 	var fetchedSignatures []*storage.Signature
-	for _, matchingReg := range matchingRegistries {
-		// FetchImageSignaturesWithRetries will try fetching of signatures with retries.
-		sigs, err := signatures.FetchImageSignaturesWithRetries(ctx, e.signatureFetcher, img, matchingReg)
-		fetchedSignatures = append(fetchedSignatures, sigs...)
-		// Skip other matching registries if we have a successful fetch of signatures, irrespective of whether
-		// signatures were found or not. Retrying this for other registries won't change the fact that signatures are
-		// available or not.
-		if err == nil {
-			break
-		}
+	for imageName, matchingRegs := range matchingRegistries {
+		for _, registry := range matchingRegs {
+			// FetchImageSignaturesWithRetries will try fetching of signatures with retries.
+			sigs, err := signatures.FetchImageSignaturesWithRetries(ctx, e.signatureFetcher, img, imageName, registry)
+			fetchedSignatures = append(fetchedSignatures, sigs...)
+			// Skip other matching registries if we have a successful fetch of signatures for the respective image name,
+			// irrespective of whether signatures were found or not.
+			// Retrying this for other registries won't change the fact that signatures are available or not for this
```

Comment:
It's not quite clear to me how one image name could have multiple matching _registries_. I can imagine that there can be multiple matching _integrations_, i.e., a global `quay.io` with some default credentials, and a more specific `quay.io/rhacs-eng` with credentials with access to this org. This would however imply that the list of registries/integrations per image name should be sorted, no?

Reply:
Yeah, its currently being used quite interchangeably within the image enricher the meaning of "registry" and "image integration" - the latter is meant here which also matches exactly the example you have given.
I've renamed the existing occurrences of e.g. functions / variables and sorted the matching image integrations by their registry hostname.